fdbserver/TLogGroup.actor.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * TLogGroup.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "fdbrpc/Replication.h"
+#include "fdbserver/TLogGroup.h"
+#include "fdbserver/WorkerInterface.actor.h"
+#include "flow/Error.h"
+#include <algorithm>
+#include <flow/UnitTest.h>
+#include <iterator>
+#include <vector>
+
+TLogGroupCollection::TLogGroupCollection(const Reference<IReplicationPolicy>& policy, int groupSize)
+  : policy(policy), GROUP_SIZE(groupSize) {}
+
+const std::vector<TLogGroupRef>& TLogGroupCollection::groups() const {
+	return recruitedGroups;
+}
+
+void TLogGroupCollection::addWorkers(const std::vector<WorkerInterface>& logWorkers) {
+	for (const auto& worker : logWorkers) {
+		recruitMap.emplace(worker.id(), TLogWorkerData::fromInterface(worker));
+	}
+}
+
+void TLogGroupCollection::recruitEverything() {
+	ASSERT_EQ(recruitMap.size() % GROUP_SIZE, 0); // FIXME
+	int numTeams = recruitMap.size() / GROUP_SIZE + (recruitMap.size() % GROUP_SIZE > 0);
+
+	std::vector<TLogGroupRef> newGroups;
+	std::unordered_set<UID> selectedServers;
+	std::vector<TLogWorkerData*> bestSet;
+
+	printf("---> Going to recruit %d groups\n", numTeams);
```

Comment:
Change to a trace event.

Reply:
Yeah, sorry about this. It was easier to debug this using `printf` right now than TraceEvent. Is supposed to be TraceEvent eventually.